ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BShellConfiguration.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell.cli;
+
+import io.ballerina.shell.Evaluator;
+import io.ballerina.shell.EvaluatorBuilder;
+import io.ballerina.shell.parser.TrialTreeParser;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Configuration that uses command utils to provide options.
+ *
+ * @since 2.0.0
+ */
+public class BShellConfiguration {
+    private final long treeParsingTimeout;
+    private final Evaluator evaluator;
+    private final InputStream inputStream;
+    private final OutputStream outputStream;
+    private boolean isDebug;
+    private boolean isDumb;
+
+    private BShellConfiguration(boolean isDebug, boolean isDumb, long treeParsingTimeout,
+                                EvaluatorMode mode, InputStream inputStream, OutputStream outputStream) {
+        this.isDebug = isDebug;
+        this.isDumb = isDumb;
+        this.treeParsingTimeout = treeParsingTimeout;
+        this.evaluator = createEvaluator(mode);
+        this.inputStream = inputStream;
+        this.outputStream = outputStream;
+    }
+
+    /**
+     * Creates and returns an evaluator based on the config.
+     *
+     * @param mode Mode to create the evaluator on.
+     * @return Created evaluator.
+     */
+    private Evaluator createEvaluator(EvaluatorMode mode) {
+        if (mode == EvaluatorMode.DEFAULT) {
```

Comment:
Can this be anything else right now?

Reply:
No, at the moment only that mode is available.

Reply:
Previously there was another mode implementation (when I was trying out different implementations) but I decided to keep the mode in because it would be easier, later on, to implement a different implementation/configurations.

Reply:
Since we have only one mode atm, IMO, we should refactor this. We can add back the checks and the enum when we are introducing more modes, WDYT? Right now, we're doing an unnecessary check that will always evaluate to true.

We can decide on this and introduce any changes separately.